### PR TITLE
Ceds 3709 is license required

### PR DESCRIPTION
**Pull request by darrylmurphy**

### CODE REVIEW

File: app/forms/declaration/IsLicenseRequired.scala [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package forms.declaration
+
+import forms.DeclarationPage
+
+object IsLicenseRequired extends DeclarationPage
```
Comment (magicaljellybeans) [inline app/forms/declaration/IsLicenseRequired.scala:21]: What's the purpose of this object?

Comment (magicaljellybeans) [reply app/forms/declaration/IsLicenseRequired.scala:21]: Ah, the backlink!

**Review by magicaljellybeans — APPROVED** [review]
Looks good to me :+1:

**Review by magicaljellybeans — DISMISSED** [review]

File: app/controllers/declaration/IsLicenseRequiredController.scala [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+      case YesNoAnswers.yes => AdditionalDocumentAddController.displayPage(_, itemId)
+      case YesNoAnswers.no if request.cacheModel.isAuthCodeRequiringAdditionalDocuments =>
+        AdditionalDocumentsController.displayPage(_, itemId)
+      case YesNoAnswers.no =>
+        AdditionalDocumentsRequiredController.displayPage(_, itemId)
+    }
+
+  private def updateCache(yesNoAnswer: YesNoAnswer, itemId: String)(implicit request: JourneyRequest[AnyContent]): Future[ExportsDeclaration] = {
+    val isLicenseRequired =
+      if (yesNoAnswer.answer == YesNoAnswers.yes) true else false
```
Comment (udinobi) [inline app/controllers/declaration/IsLicenseRequiredController.scala:85]: What about just having...
```
val isLicenseRequired = yesNoAnswer.answer == YesNoAnswers
```

**Comment by udinobi** [comment]
Is the new page already included in `docs/Customs Declare Exports AutoComplete.js` ?

File: app/controllers/navigation/Navigator.scala [hunk 3]
```diff
@@ -406,16 +423,36 @@ class Navigator @Inject()(
...
-  private def additionalDocumentsPreviousPage(cacheModel: ExportsDeclaration, mode: Mode, itemId: String): Call =
+  private def isLicenseRequiredPreviousPage(cacheModel: ExportsDeclaration, mode: Mode, itemId: String): Call =
+    if (cacheModel.listOfAdditionalInformationOfItem(itemId).nonEmpty)
+      routes.AdditionalInformationController.displayPage(mode, itemId)
+    else
+      routes.AdditionalInformationRequiredController.displayPage(mode, itemId)
+
+  private def additionalDocumentsClearancePreviousPage(cacheModel: ExportsDeclaration, mode: Mode, itemId: String): Call =
     if (cacheModel.listOfAdditionalDocuments(itemId).nonEmpty)
       routes.AdditionalDocumentsController.displayPage(mode, itemId)
```
Comment (magicaljellybeans) [inline app/controllers/navigation/Navigator.scala:451]: Not your logic, but I don't get this behaviour - if there are no documents it displays /add-additional-documentation again? :confused:

File: app/controllers/navigation/Navigator.scala [hunk 4]
```diff
@@ -406,16 +423,36 @@ class Navigator @Inject()(
...
+    if (cacheModel.isAuthCodeRequiringAdditionalDocuments || isLicenseRequired) routes.IsLicenseRequiredController.displayPage(mode, itemId)
+    else routes.AdditionalDocumentsRequiredController.displayPage(mode, itemId)
+
+  }
+
+  private def additionalDocumentsSummaryClearancePreviousPage(cacheModel: ExportsDeclaration, mode: Mode, itemId: String): Call =
+    if (cacheModel.listOfAdditionalInformationOfItem(itemId).nonEmpty)
       routes.AdditionalInformationController.displayPage(mode, itemId)
     else
       routes.AdditionalInformationRequiredController.displayPage(mode, itemId)
```
Comment (magicaljellybeans) [inline app/controllers/navigation/Navigator.scala:441]: Wouldn't the previous page from additional documents summary just be /additional-documentation?

File: app/controllers/declaration/IsLicenseRequiredController.scala [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+            navigator.continueTo(mode, nextPage(yesNo, itemId))
+        }
+      )
+  }
+
+  private def nextPage(yesNoAnswer: YesNoAnswer, itemId: String)(implicit request: JourneyRequest[AnyContent]): Mode => Call =
+    yesNoAnswer.answer match {
+      case YesNoAnswers.yes => AdditionalDocumentAddController.displayPage(_, itemId)
+      case YesNoAnswers.no if request.cacheModel.isAuthCodeRequiringAdditionalDocuments =>
+        AdditionalDocumentsController.displayPage(_, itemId)
```
Comment (magicaljellybeans) [inline app/controllers/declaration/IsLicenseRequiredController.scala:78]: I think this should be `AdditionalDocumentAddController.displayPage(_, itemId)`?

File: app/controllers/navigation/Navigator.scala (hunk 4 above)
Comment (darrylmurphy) [reply app/controllers/navigation/Navigator.scala:441]: I'm maintaining the same logic as what came before this story here, so I'm hesitant to fiddle with the back button any further!

File: app/controllers/navigation/Navigator.scala (hunk 3 above)
Comment (darrylmurphy) [reply app/controllers/navigation/Navigator.scala:451]: If there are documents it shows the list

File: app/controllers/declaration/AdditionalInformationRequiredController.scala [hunk 6]
```diff
@@ -69,10 +70,11 @@ class AdditionalInformationRequiredController @Inject()(
 
   private def form: Form[YesNoAnswer] = YesNoAnswer.form(errorKey = "declaration.additionalInformationRequired.error")
 
-  private def nextPage(yesNoAnswer: YesNoAnswer, itemId: String): Mode => Call =
+  private def nextPage(yesNoAnswer: YesNoAnswer, itemId: String)(implicit request: JourneyRequest[_]): Mode => Call =
     yesNoAnswer.answer match {
-      case YesNoAnswers.yes => AdditionalInformationController.displayPage(_, itemId)
-      case YesNoAnswers.no  => AdditionalDocumentsController.displayPage(_, itemId)
+      case YesNoAnswers.yes                                                        => AdditionalInformationController.displayPage(_, itemId)
+      case YesNoAnswers.no if request.declarationType == DeclarationType.CLEARANCE => AdditionalDocumentsController.displayPage(_, itemId)
```
[the remark this answers is not in the document]
Comment (darrylmurphy) [reply app/controllers/declaration/AdditionalInformationRequiredController.scala:76]: AdditionalDocumentsController.displayPage() does that work for us

**Review by magicaljellybeans — APPROVED** [review]
Thanks for addressing comments, looks ship-shape to me :+1:

**Review by magicaljellybeans — APPROVED** [review]

**Pull request merged**